### ISSUE TITLE
retry function in test.sh doesn't seem to be retrying

### ISSUE THREAD

**Issue by sublimator** [coilhq/web-monetization-projects#2544]
```
#!/bin/bash -u
# We use set -e and bash with -u to bail on first non zero exit code of any
# processes launched or upon any unbound variable.
# We use set -x to print commands before running them to help with
# debugging.
set -ex
```
Probably due to the flags we are using

**Comment by sublimator** [comment]
Altered test.sh to instead run another yarn script that simply exited with code 1, and test.sh retried 3 times successfully.
So it's unclear why the loop stops at `((n++))`

**Comment by sublimator** [comment]
![image](https://user-images.githubusercontent.com/525211/150886491-49583dfb-5824-49ee-8fc3-5d7413c5ef82.png)

**Comment by sublimator** [comment]
Seems bash with `-u` flag would bail on non-zero "exit" of ((n++)) statement

**Comment by sublimator** [comment]
Fixed in #2547
